sub/dec_sub.c

```diff
@@ -210,6 +212,7 @@ static void update_segment(struct dec_sub *sub)
             talloc_free(sub->sd);
             sub->sd = new;
             update_subtitle_speed(sub);
+            sub_control(sub, SD_CTRL_SET_TOP, &(bool){!!sub->order});
```

Comment:
Should this really be a bool? Allegedly `get_order` can also return -1. I'm not sure what exactly happens in that case.

Reply:
The function that constructs `track->d_sub` is `init_subdec` which is only called when a sub is selected (otherwise `reinit_sub`, which calls `init_subdec`, returns early), so sub->order is always 0 or 1.

Reply:
Cool. Can this be `sub->order` instead of `!!sub->order`? Is that the same?

Reply:
Actually that whole cast was totally pointless since `sub/sd_ass.c:control` already casts `arg` to `bool` when `cmd` is `SD_CTRL_SET_TOP`. I removed it in the other `sub_control` cast as well.